test/e2e/e2e.go

```diff
@@ -0,0 +1,202 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"path"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/ginkgo/config"
+	"github.com/onsi/ginkgo/reporters"
+	"github.com/onsi/gomega"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/unversioned/clientcmd"
+	"k8s.io/kubernetes/pkg/cloudprovider"
+	gcecloud "k8s.io/kubernetes/pkg/cloudprovider/providers/gce"
+	"k8s.io/kubernetes/pkg/util"
+)
+
+const (
+	// podStartupTimeout is the time to allow all pods in the cluster to become
+	// running and ready before any e2e tests run. It includes pulling all of
+	// the pods (as of 5/18/15 this is 8 pods).
+	podStartupTimeout = 10 * time.Minute
+)
+
+var (
+	cloudConfig = &testContext.CloudConfig
+
+	reportDir = flag.String("report-dir", "", "Path to the directory where the JUnit XML reports should be saved. Default is empty, which doesn't generate these reports.")
+)
+
+type E2ETests struct {
+}
+
+func NewE2ETests() *E2ETests {
+	tests := &E2ETests{}
+	tests.RegisterE2EFlags()
+	return tests
+}
+
+// Sanity checks that the e2e's have the basic information required to run before launching them.
+func (e2e *E2ETests) IsSetup() bool {
+	if &testContext.KubeConfig == nil || *&testContext.KubeConfig == "" {
+		glog.Warningf("KubeConfig value not configured.")
+		return false
+	}
+
+	// Add more checks below...
+	return true
+}
+
+func (e2e *E2ETests) RegisterE2EFlags() {
```

Comment:
I've put this on the struct because over time we want the struct  to hold more info, and this allows us to contextualize the fact that all this method does is make sure the configuration params are written to an object.  

Right now that object is external to the e2e struct,  but in general there is no reason to register these flags outside the context of an e2e struct.

Let me know if that justifies it to you. If not I can make it a global function but I think this helps to move us away from global config dependencies , and towards a purely data-coupled configuration flow.